src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -878,25 +878,14 @@ HeapWord* ShenandoahHeap::allocate_memory(ShenandoahAllocRequest& req) {
     // It might happen that one of the threads requesting allocation would unblock
     // way later after GC happened, only to fail the second allocation, because
     // other threads have already depleted the free storage. In this case, a better
-    // strategy is to try again, as long as GC makes progress.
-    //
-    // Then, we need to make sure the allocation was retried after at least one
-    // Full GC, which means we want to try more than ShenandoahFullGCThreshold times.
-
-    size_t tries = 0;
-
-    while (result == nullptr && _progress_last_gc.is_set()) {
-      tries++;
-      control_thread()->handle_alloc_failure(req);
-      result = allocate_memory_under_lock(req, in_new_region);
-    }
-
-    while (result == nullptr && tries <= ShenandoahFullGCThreshold) {
```

Comment:
Just want to point out, the description of `ShenandoahFullGCThreshold` does not cover this use case (also, the original code here was not concerned with full GCs):
>           "How many back-to-back Degenerated GCs should happen before going to a Full GC."